code/game/machinery/doors/door.dm

```diff
@@ -292,10 +292,18 @@ var/list/all_doors = list()
 
 /obj/machinery/door/Cross(atom/movable/mover, turf/target, height=1.5, air_group = 0)
 	if(air_group) return 0
-	if(istype(mover) && mover.checkpass(PASSGLASS))
-		return !opacity
+	if(istype(mover))
+		if(mover.checkpass(PASSGLASS))
+			return !opacity
+		if(mover.checkpass(PASSDOOR))
+			return 1
 	return !density
 
+/obj/machinery/door/Crossed(AM as mob|obj) //Since we can't actually quite open AS the car goes through us, we'll do the next best thing: open as the car goes into our tile.
+	if(istype(AM, /obj/structure/bed/chair/vehicle/wizmobile)) //Which is not 100% correct for things like windoors but it's close enough.
+		open()
```

Comment:
Nope, neither does it open welded doors, it just noclips through them (which was ostensibly intended)